python/lbann/contrib/launcher.py

```diff
@@ -68,9 +107,14 @@ def make_batch_script(*args, **kwargs):
 
     """
 
+    print('Starting to make the batch script')
     # Livermore Computing
-    if lbann.contrib.lc.systems.is_lc_system():
+    if _center == LC and lbann.contrib.lc.systems.is_lc_system():
```

Comment:
Perhaps we want to "template" this:
```python
if _center == LC:
  import lbann.contrib.lc.launcher as the_laucher
  import lbann.contrib.lc.systems as the_systems
elif _center == NERSC:
  import lbann.contrib.nersc.launcher as the_launcher
  import lbann.contrib.nersc.systesm as the_systems
else:
  # Whatever fallback

if the_systems.is_known_system():
  return the_launcher.make_batch_script(*args, **kwargs)
else
  return default_launcher.make_batch_script(*args, **kwargs)
```
Just because it's Python doesn't mean we shouldn't care about good, clean programming practice. `if...else` is rarely clean, and increasingly rarely necessary if all you want to do is change a few characters among the cases.

Also, arguably, the default launcher should be `mpiexec`, since that's what standards-recommended for MPI implementations. (Note, it is a recommendation, not a standards requirement. But the (non-)words "Slurm", "srun", "LSF", and "jsrun" do not appear in the standard at all, so in some sense, "mpiexec" is a better guess than any of them ("mpirun" is not considered portable by the standard).

Reply:
That's a nice pattern, I'll implement it.

Also, I would agree the basic MPI launcher would be the most natural default.

Reply:
I agree, then @benson31 can test it on his system.